services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -271,6 +270,10 @@
     "CrossRegionAccessEnabled":{
       "documentation":"Enables cross-region bucket access for this client",
       "type":"boolean"
+    },
+    "multipartEnabled":{
```

Comment:
This also adds the multipart-enable config and method to the sync version of the client, as it adds the method the `S3BaseClientBuilder` interface which is used by both the sync and async builders:
```
S3Client client = S3Client.builder()
    .multipartEnabled(true) // <--- should not be available on sync builder
    .build();
```
I think we should avoid being able to define this config on the sync client and keep it on the async client only